django/db/migrations/graph.py

```diff
@@ -322,7 +249,7 @@ def leaf_nodes(self, app=None):
                 leaves.add(node)
         return sorted(leaves)
 
-    def ensure_not_cyclic(self, start, get_children):
```

Comment:
Hmmm. `start` was unused anyway...

Reply:
Yes, I've noticed this too (see the commit message). Thanks to this, we can check for existence of cycles once, instead of doing this for every `start`.